app/experimenter/static/js/components/TimelinePopForm.js

```diff
@@ -0,0 +1,372 @@
+import PropTypes from "prop-types";
+import React from "react";
+import { Button, Container, Row, Col } from "react-bootstrap";
+import { boundClass } from "autobind-decorator";
+import { fromJS, Map } from "immutable";
+
+import DesignInput from "experimenter/components/DesignInput";
+import LabeledMultiSelect from "experimenter/components/LabeledMultiSelect";
+
+import { makeApiRequest } from "experimenter/utils/api";
+import {
+  VERSION_CHOICES,
+  PROPOSED_START_DATE_HELP,
+  PROPOSED_DURATION_HELP,
+  PROPOSED_ENROLLMENT_HELP,
+  CHANNEL_HELP,
+  POPULATION_PERCENT_HELP,
+  VERSION_HELP,
+  PLATFORM_HELP,
+  CLIENT_MATCHING_HELP,
+  COUNTRIES_LOCALES_HELP,
+} from "experimenter/components/constants";
+
+@boundClass
+class TimelinePopForm extends React.PureComponent {
+  static propTypes = {
+    slug: PropTypes.string,
+    shouldHavePopPercent: PropTypes.string,
+    allCountries: PropTypes.array,
+    allLocales: PropTypes.array,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: new Map(),
+      errors: new Map(),
+      loaded: false,
+      saving: false,
+    };
+  }
+
+  getEndpointUrl() {
+    return `experiments/${this.props.slug}/timeline-population/`;
+  }
+
+  async componentDidMount() {
+    try {
+      const data = await makeApiRequest(this.getEndpointUrl());
+      this.setState({
+        loaded: true,
+        data: fromJS(data),
+      });
+    } catch (error) {
+      console.error(error.message);
+    }
+  }
+
+  handleDataChange(key, value) {
+    if (value == "") {
+      value = null;
+    }
+    this.setState(({ data }) => ({
+      data: data.set(key, value),
+    }));
+  }
+
+  handleMultiSelectDataChange(selectedOptions, name) {
+    if (selectedOptions == null) {
+      selectedOptions = [];
+    }
+    this.setState(({ data }) => ({
+      data: data.set(name, fromJS(selectedOptions)),
+    }));
+  }
+
+  displayPopulationPercent() {
+    if (this.props.shouldHavePopPercent == "True") {
```

Comment:
1) use `===` comparisons
2) Is there a way to do this that uses the actual boolean value and not a string representation of a Python bool?

Reply:
I addressed 1) elsewhere.

2) yes there was a way to do it.. in `edit_timeline_population.html`, I elected not to pass a value to the div if should_have_population_percentage was false, thus allowing the JS in TimelinePopForm to read:

`if (this.props.shouldHavePopPercent) {...}`

Reply:
Yeah but isn't this case different for the equality comparison than the null case?  You do want the `===` strictness here?

Also that seems good to me, why did you switch it to this?

Reply:
yes yes, you are right, the `===` still needs to be used if i left it as a string.

I switched it to the above way because it seemed better than having the `this.props.shouldHavePopPercent == "True"` stuff.